component/src/test/java/org/wso2/extension/siddhi/io/jms/source/JMSSourceTestCase.java

```diff
@@ -228,22 +224,226 @@ public void testJMSTopicSource4() throws InterruptedException {
 
         // deploying the execution plan
         SiddhiManager siddhiManager = new SiddhiManager();
-        String inStreamDefinition = "" +
-                "@source(type='jms', @map(type='xml'), "
+        String inStreamDefinition = "" + "@source(type='jms', @map(type='xml'), "
                 + "provider.url='vm://localhost',"
-                + "destination='DAS_JMS_TEST', "
+                + "destination='DAS_JMS_TEST', " + "connection.factory.type='topic',"
+                + "connection.factory.jndi.name='TopicConnectionFactory',"
+                + "transport.jms.SubscriptionDurable='true', "
+                + "transport.jms.DurableSubscriberClientID='wso2dasclient1'" + ")"
+                + "define stream inputStream (name string, age int, country string);";
+        String query = ("@info(name = 'query1') "
+                + "from inputStream "
+                + "select *  "
+                + "insert into outputStream;");
+        SiddhiAppRuntime executionPlanRuntime = siddhiManager.createSiddhiAppRuntime(inStreamDefinition + query);
+
+        executionPlanRuntime.addCallback("query1", new QueryCallback() {
+            @Override
+            public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
+                EventPrinter.print(timeStamp, inEvents, removeEvents);
+                for (Event event : inEvents) {
+                    eventCount.incrementAndGet();
+                    receivedEventNameList.add(event.getData(0).toString());
+                }
+            }
+        });
+        executionPlanRuntime.start();
+        siddhiManager.shutdown();
+    }
+
+    @Test
+    public void testJMSTopicSourcePause() throws InterruptedException {
```

Comment:
It is always better to add a comment on what each test case is doing? It will be very help-ful in long run